src/storage/v2/replication/replication_client.cpp

```diff
@@ -29,14 +28,27 @@ namespace memgraph::storage {
 ReplicationStorageClient::ReplicationStorageClient(::memgraph::replication::ReplicationClient &client)
     : client_{client} {}
 
-void ReplicationStorageClient::CheckReplicaState(Storage *storage) {
+void ReplicationStorageClient::UpdateReplicaState(Storage *storage, std::any gk) {
   uint64_t current_commit_timestamp{kTimestampInitialId};
 
   auto &replStorageState = storage->repl_storage_state_;
-  auto stream{client_.rpc_client_.Stream<replication::HeartbeatRpc>(
-      storage->id(), replStorageState.last_commit_timestamp_, std::string{replStorageState.epoch_.id()})};
+  replication::HeartbeatRes replica;
+
+  std::optional<rpc::Client::StreamHandler<replication::HeartbeatRpc>> hb_stream{};
+
+  hb_stream.emplace(client_.rpc_client_.Stream<replication::HeartbeatRpc>(
+      storage->uuid(), replStorageState.last_commit_timestamp_, std::string{replStorageState.epoch_.id()}));
+  replica = hb_stream->AwaitResponse();
```

Comment:
What?! AFAICT no need for optional stream + HeartbeatRes assignment

Reply:
Sorry, left from an older attempt at recovery